android/src/main/java/com/oney/WebRTCModule/WebRTCModule.java

```diff
@@ -417,29 +415,30 @@ public void peerConnectionInit(ReadableMap configuration, int id) {
     }
 
     MediaStream getStreamForReactTag(String streamReactTag) {
+        // This function _only_ gets called from WebRTCView, in the UI thread.
+        // Hence make sure we run this code in the executor or we run at the risk
+        // of being out of sync.
         try {
-            return ThreadUtils.runOnExecutorAndWait(() -> getStreamForReactTagAsync(streamReactTag));
-        } catch (Exception e) {
-            Log.d(TAG, "getStreamForReactTag() failed");
-            e.printStackTrace();
-        }
-        return null;
-    }
+            return (MediaStream) ThreadUtils.submitToExecutor((Callable<Object>) () -> {
+                MediaStream stream = localStreams.get(streamReactTag);
 
-    MediaStream getStreamForReactTagAsync(String streamReactTag) {
-        MediaStream stream = localStreams.get(streamReactTag);
-
-        if (stream == null) {
-            for (int i = 0, size = mPeerConnectionObservers.size(); i < size; i++) {
-                PeerConnectionObserver pco = mPeerConnectionObservers.valueAt(i);
-                stream = pco.remoteStreams.get(streamReactTag);
```

Comment:
these changes came in from the second merge commit: https://github.com/daily-co/react-native-webrtc/pull/17/commits/1217b35e6a7be37bf0da08b5b68acb37d3040c70

but seem odd since you remove a whole function... can you expand on these changes?  again... i'm totally out of my element

Reply:
The WebRTCModule.java is one of the classes that we have changed, to support basically this features:
- Return a value from the method `mediaStreamTrackSwitchCamera`
- Creating a new method `mediaStreamTrackGetCameraFacingMode`
- Creating a new method `setDailyAudioMode`

When I did the merge, that class had a lot of changes between the version 1.89.4 and this version 1.94.1. So, for some reason, I didnt merge right those parts that you have described above. That is why I needed to create a new commit, to remove the parts that have been wrong merged. Those parts were not supposed to exist in this version 1.94.1.

You can see here the comparative between how is this branch now and the upstream from react-native, you will see that only have the new features that I have described:
https://github.com/react-native-webrtc/react-native-webrtc/compare/master...daily-co:main-synced-to-1.94.1#diff-6732b1c8f20f6bb8141dcabe3d92b5d92afe0cbc8ce806d3dd7d68bfe9cb26c3